### PR TITLE
Add note to pom warning that care should be taken to align versions when upgrading strimzi/fabric8/sundr

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

_Select the type of your PR_

- Documentation

### Description

Add note to pom warning that care should be taken to align versions when upgrading strimzi/fabric8/sundr.

I'm being cautious.  There are probably cases when we should overrule this stipulation (CVE remediation being one case)

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonar-Lint warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -63,7 +62,11 @@
 
         <!-- Test dependencies -->
         <strimzi.version>0.37.0</strimzi.version>
+        <!-- keep Fabric8 version in lock step with the Fabric8 version used by Strimzi -->
         <fabric8.kubernetes.version>6.8.1</fabric8.kubernetes.version>
+        <!-- keep Sundr version in lock step with the Sundr version used by Strimzi and Fabric8 -->
```
Comment (SamBarker) [inline pom.xml:67]: Where does the dependency on Strimzi come from? I can only see the version being defined but never used...

**Review by SamBarker — APPROVED** [review]
The comment gives me the warning about whats needed but I miss the context of why/what goes wrong if they move independently. 

I'm also curious as to why our build doesn't catch the issues. If the build works prior to merging in why shouldn't they move independently? If there are still issues how can we catch them in an automated fashion?

File: pom.xml (hunk 1 above)
Comment (robobario) [reply pom.xml:67]: kroxylicious-systemtests pulls in io.strimzi:api

Comment (SamBarker) [reply pom.xml:67]: Should we be excluding sundr from io.strimzi:api then?

**Review by franvila — APPROVED** [review]

**Comment by k-wall** [comment]
As described above, I now think this is the wrong solution.

**Pull request closed without merging**